CaloCluster/src/CaloProtoClusterFromCrystalHit_module.cc

```diff
@@ -60,79 +61,183 @@ namespace mu2e {
 
   public:
 
-    typedef std::vector<const CaloCrystalHit*>  CaloCrystalVec;
-    typedef std::list<const CaloCrystalHit*>    CaloCrystalList;
-
-
-    explicit CaloProtoClusterFromCrystalHit(fhicl::ParameterSet const& pset) :
-      art::EDProducer{pset},
-      caloCrystalToken_{consumes<CaloCrystalHitCollection>(pset.get<std::string>("caloCrystalModuleLabel"))},
-      producerNameMain_(pset.get<std::string>("mainClusterCollName")),
-      producerNameSplit_(pset.get<std::string>("splitClusterCollName")),
-      EminSeed_(pset.get<double>("EminSeed")),
-      EnoiseCut_(pset.get<double>("EnoiseCut")),
-      ExpandCut_(pset.get<double>("ExpandCut")),
-      timeCut_(pset.get<double>("timeCut")),
-      deltaTime_(pset.get<double>("deltaTime")),
-      diagLevel_(pset.get<int>("diagLevel",0)),
-      messageCategory_("CLUSTER")
-    {
-      produces<CaloProtoClusterCollection>(producerNameMain_);
-      produces<CaloProtoClusterCollection>(producerNameSplit_);
-    }
-
-    void produce(art::Event& e) override;
-
-  private:
-
-    art::ProductToken<CaloCrystalHitCollection> const caloCrystalToken_;
-    std::string       producerNameMain_;
-    std::string       producerNameSplit_;
-    double            EminSeed_;
-    double            EnoiseCut_;
-    double            ExpandCut_;
-    double            timeCut_;
-    double            deltaTime_;
-    int               diagLevel_;
-    const std::string messageCategory_;
-
-    void makeProtoClusters(CaloProtoClusterCollection& caloProtoClustersMain,
-                           CaloProtoClusterCollection& caloProtoClustersSplit,
-                           const art::Handle<CaloCrystalHitCollection>& CaloCrystalHitsHandle);
-
-    void filterByTime(CaloCrystalList& liste,
-                      const std::vector<double>& clusterTime,
-                      std::list<const CaloCrystalHit*> &seedList);
-
-    void fillCluster(CaloProtoClusterCollection& caloProtoClustersColl, const CaloCrystalList& clusterList,
-                     const art::Handle<CaloCrystalHitCollection>& CaloCrystalHitsHandle);
-
-    void dump(const std::vector<CaloCrystalList>& caloIdHitMap);
-
-  };
+	typedef std::vector<const CaloCrystalHit*>  CaloCrystalVec;
+	typedef std::list<const CaloCrystalHit*>    CaloCrystalList;
+
+
+	explicit CaloProtoClusterFromCrystalHit(fhicl::ParameterSet const& pset) :
+	art::EDProducer{pset},
+	caloCrystalToken_{consumes<CaloCrystalHitCollection>(pset.get<std::string>("caloCrystalModuleLabel"))},
+	producerNameMain_(pset.get<std::string>("mainClusterCollName")),
+	producerNameSplit_(pset.get<std::string>("splitClusterCollName")),
+	isOnline_(pset.get<bool>("isOnline",false)),
+	minEnergy_(pset.get<double>("minEnergy",0)),
```

Comment:
this parameter refers to the minimum cluster energy. I would consider calling it in a more meaningful way, like minClusterEnergy_